### PR TITLE
Use rn typeface utils for font loading

### PR DESCRIPTION
**Pull request by danilobuerger**
This will get us all the new android rn typeface features like using `fontWeight` and `fontStyle` everywhere. Also we will be able to use font xmls as explained here: https://developer.android.com/guide/topics/ui/look-and-feel/fonts-in-xml

### CODE REVIEW

**Comment by danilobuerger** [comment]
@guyca Should we expand this PR to also expose these props at the js side? If yes, we would also need to catch up on iOS 🤔

**Comment by guyca** [comment]
@danilobuerger We should document the newly supported options - `fontWeight` and `fontStyle` in the following places:
* [bottomTab](https://wix.github.io/react-native-navigation/api/options-bottomTab#fontfamily)
* [title](https://wix.github.io/react-native-navigation/api/options-title#fontfamily)
* [subtitle](https://wix.github.io/react-native-navigation/api/options-subtitle)
* [button](https://wix.github.io/react-native-navigation/api/options-button)

Have you checked if the TypeScript declarations need to be updated?

Regarding iOS - lets do any additional iOS work in a different PR so it's easier to land. WDYT?

**Comment by danilobuerger** [comment]
@guyca Updated docs and typings.

**Review by guyca — CHANGES_REQUESTED** [review]
As discussed, all that's left is to update the docs and the TS declarations. iOS support will be added later in a separate PR.
Amazing work @danilobuerger 🙇 🙌

File: lib/src/interfaces/Options.ts [hunk 1]
```diff
@@ -167,12 +181,17 @@ export interface OptionsTopBarLargeTitle {
    */
   color?: Color;
   /**
-   * Set the font family of large title's text
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:184]: Out of curiosity, why was this description changed? Seems like the previous phrasing was more explicit.

**Review by guyca — APPROVED** [review]

File: lib/src/interfaces/Options.ts (hunk 1 above)
Comment (danilobuerger) [reply lib/src/interfaces/Options.ts:184]: Copy & paste hack job 😢. I copied the `fontFamily`, `fontStyle` and `fontWeight` to every interface that needed it and didn't circle back to change the comments. Do you want that changed?

Comment (guyca) [inline lib/src/interfaces/Options.ts:184]:
```suggestion
   * Set the font family of the large title text
```

File: lib/src/interfaces/Options.ts [hunk 2]
```diff
@@ -167,12 +181,17 @@ export interface OptionsTopBarLargeTitle {
...
    */
   fontFamily?: FontFamily;
   /**
-   * Set the font weight, ignore fontFamily and use the iOS system fonts instead
-   * #### (iOS specific)
+   * Set the font style for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:188]:
```suggestion
   * Set the font style of the large title text
```

File: lib/src/interfaces/Options.ts [hunk 3]
```diff
@@ -191,14 +210,17 @@ export interface OptionsTopBarTitle {
    */
   color?: Color;
   /**
-   * Title font family
-   *
-   * Make sure that the font is available
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:213]:
```suggestion
   * Set the font family for the title
```

File: lib/src/interfaces/Options.ts [hunk 4]
```diff
@@ -191,14 +210,17 @@ export interface OptionsTopBarTitle {
...
    */
   fontFamily?: FontFamily;
   /**
-   * Set the font weight, ignore fontFamily and use the iOS system fonts instead
-   * #### (iOS specific)
+   * Set the font style for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:217]:
```suggestion
   * Set the font style for the title
```

File: lib/src/interfaces/Options.ts [hunk 5]
```diff
@@ -248,12 +270,17 @@ export interface OptionsTopBarSubtitle {
    */
   color?: Color;
   /**
-   * Set subtitle font family
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:273]:
```suggestion
   * Set the font family for the subtitle
```

File: lib/src/interfaces/Options.ts [hunk 6]
```diff
@@ -248,12 +270,17 @@ export interface OptionsTopBarSubtitle {
...
    */
   fontFamily?: FontFamily;
   /**
-   * Set the font weight, ignore fontFamily and use the iOS system fonts instead
-   * #### (iOS specific)
+   * Set the font style for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:277]:
```suggestion
   * Set the font style for the subtitle
```

File: lib/src/interfaces/Options.ts [hunk 7]
```diff
@@ -301,9 +328,19 @@ export interface OptionsTopBarBackButton {
    */
   fontSize?: number;
   /**
-   * Set subtitle font family
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:331]:
```suggestion
   * Set the font family for the back button
   * #### (iOS specific)
```

File: lib/src/interfaces/Options.ts [hunk 8]
```diff
@@ -397,12 +434,17 @@ export interface OptionsTopBarButton {
    */
   accessibilityLabel?: string;
   /**
-   * Set the button font family
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:437]:
```suggestion
   * Set the font family for the button's text
```

File: lib/src/interfaces/Options.ts [hunk 9]
```diff
@@ -397,12 +434,17 @@ export interface OptionsTopBarButton {
...
    */
-  fontFamily?: string;
+  fontFamily?: FontFamily;
   /**
-   * Set the font weight, ignore fontFamily and use the iOS system fonts instead
-   * #### (iOS specific)
+   * Set the font style for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:441]:
```suggestion
   * Set the font style for the button's text
```

File: lib/src/interfaces/Options.ts [hunk 10]
```diff
@@ -743,12 +785,17 @@ export interface OptionsBottomTab {
    */
   selectedTextColor?: Color;
   /**
-   * Set the text font family
+   * Set the font family for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:788]:
```suggestion
   * Set the font family for the tab's text
```

File: lib/src/interfaces/Options.ts [hunk 11]
```diff
@@ -743,12 +785,17 @@ export interface OptionsBottomTab {
...
    */
   fontFamily?: FontFamily;
   /**
-   * Set the font weight, ignore fontFamily and use the iOS system fonts instead
-   * #### (iOS specific)
+   * Set the font style for a text
```
Comment (guyca) [inline lib/src/interfaces/Options.ts:792]:
```suggestion
   * Set the font style for the tab's text
```

**Review by guyca — CHANGES_REQUESTED** [review]

**Review by guyca — DISMISSED** [review]

**Review by guyca — APPROVED** [review]

**Pull request merged**